pkg/agents/team5/agent.go

```diff
@@ -86,6 +93,114 @@ func (a *CustomAgent5) checkForLeader() {
 	if diceRoll >= a.leadership {
 		a.Log("An agent has become a leader", infra.Fields{"dice roll": diceRoll, "leadership": a.leadership, "selfishness": a.selfishness, "floor": a.Floor()})
 		//TODO: Send treaties here about eating less food
+		hpLevel := a.currentAimHP - ((a.currentAimHP-a.HealthInfo().WeakLevel)/10)*(diceRoll-a.leadership)
+		a.currentProposal = messages.NewTreaty(messages.HP, hpLevel, messages.LeavePercentFood, 100, messages.GE, messages.EQ, 5, a.ID())
+		a.treatySendCounter = 1
+		a.Log("Agent is sending a treaty proposal", infra.Fields{"Proposed Max Hp": hpLevel})
+	}
+}
+
+func (a *CustomAgent5) statementsIntersect(op1 messages.Op, value1 int, op2 messages.Op, value2 int) bool {
+	switch op1 {
+	case messages.EQ:
+		return ((op2 == messages.EQ && value1 == value2) ||
+			(op2 == messages.LT && value1 < value2) ||
+			(op2 == messages.LE && value1 <= value2) ||
+			(op2 == messages.GT && value1 > value2) ||
+			(op2 == messages.GE && value1 >= value2))
+	case messages.LT:
+		return ((op2 == messages.EQ && value1 < value2) ||
+			op2 == messages.LT ||
+			op2 == messages.LE ||
+			(op2 == messages.GT && (value1-1) > value2) || // need the -1 here to disallow x<3 AND x>2
+			(op2 == messages.GE && value1 > value2))
+	case messages.LE:
+		return (op2 == messages.LT ||
+			op2 == messages.LE ||
+			value1 > value2)
```

Comment:
More checks needed here
op2 == EQ && value1 >= value2
op2 == GT && value1 > value2
op2 == GE && value1 >= value2

These should all return true

Reply:
yep, not sure how I missed that, thanks!